### PR TITLE
enhancement(aws_s3 sink): emit processed bytes when sending requests

### PR DESCRIPTION
**Pull request by tobz**
We weren't shipping this before.

Signed-off-by: Toby Lawrence <toby@nuclearfurnace.com>

### CODE REVIEW

File: src/sinks/aws_s3.rs [hunk 1]
```diff
@@ -298,6 +290,10 @@ impl Service<Request> for S3Sink {
         }
         let tagging = tagging.finish();
 
+        emit!(S3EventsSent {
+            byte_size: request.body.len(),
+        });
```
Comment (jszwedko) [inline src/sinks/aws_s3.rs:295]: 🤔 I think we typically don't emit these metrics unless the sending is successful.

Comment (tobz) [reply src/sinks/aws_s3.rs:295]: In practice, we do... which I don't agree with.  I'm happy to shift it to a position where it is more representative of where/when it leaves Vector, but was initially going for consistency.

(As an example, the Elasticsearch sink emits this metric simply when the log event is encoded, and it happens similarly in the generic HTTP sink, etc.)

Comment (jszwedko) [reply src/sinks/aws_s3.rs:303]: Indeed, you are right. I think this is OK for now then to maintain consistency.

**Review by jszwedko — APPROVED** [review]

File: src/sinks/aws_s3.rs (hunk 1 above)
Comment (spencergilbert) [reply src/sinks/aws_s3.rs:303]: Yeah, I ran into similar problems trying to instrument some other sinks and being unsure where to actually emit these

**Review by spencergilbert — APPROVED** [review]

**Pull request merged**